test/UriTest.php

```diff
@@ -87,6 +87,14 @@ public function testWithUserInfoReturnsNewInstanceIfUserAndPasswordAreSameAsBefo
         $this->assertEquals('https://user:pass@local.example.com:3001/foo?bar=baz#quz', (string) $new);
     }
 
+    public function testWithUserInfoEncodesUsernameAndPassword()
+    {
+        $uri = new Uri('https://user:pass@local.example.com:3001/foo?bar=baz#quz');
+        $new = $uri->withUserInfo('foo:bar', 'baz:bat');
+
+        $this->assertSame('foo%3Abar:baz%3Abat', $new->getUserInfo());
+    }
```

Comment:
Probably either a second assertion, or add `%25` to the first input to cover that it's not getting accepted as urlencoded input.

```
public function testWithUserInfoDoesNotExpectUrlencodedCharactersInUsernameAndPassword()
{
    $uri = new Uri('https://localhost');
    $new = $uri->withUserInfo('%25', '%25');

   $this->assertSame('%2525:%2525', $new->getUserInfo());
}
```